tests/bundle/test_bundle.py

```diff
@@ -131,3 +131,43 @@ def test_max_allowed_ops_staked_sender(w3, entrypoint_contract, helper_contract)
         params=[ophash],
     ).send()
     assert response.result["userOpHash"] == ophash
+
+
+@pytest.mark.parametrize("mode", ["manual"], ids=[""])
+@pytest.mark.usefixtures("clear_state", "set_bundling_mode")
+def test_ban_user_op_access_other_ops_sender_in_bundle(w3, entrypoint_contract, helper_contract):
+    # wallet 2 will treat this wallet as a "token" and access associated storage
+    wallet1_token = deploy_and_deposit(w3, entrypoint_contract, "TestFakeWalletToken", False)
+    wallet2 = deploy_and_deposit(w3, entrypoint_contract, "TestFakeWalletToken", False)
+    wallet1_token.functions.sudoSetBalance(wallet1_token.address, 10**18).transact(
+        {"from": w3.eth.accounts[0]}
+    )
+    wallet1_token.functions.sudoSetBalance(wallet2.address, 10**18).transact(
+        {"from": w3.eth.accounts[0]}
+    )
+    wallet2.functions.sudoSetAnotherWallet(wallet1_token.address).transact(
+        {"from": w3.eth.accounts[0]}
+    )
+    calldata1 = wallet2.address
+    calldata2 = "0x"
+    user_op1 = UserOperation(sender=wallet1_token.address, callData=calldata1)
+    user_op2 = UserOperation(sender=wallet2.address, callData=calldata2)
+    user_op1.send()
+    user_op2.send()
+    send_bundle_now()
+
```

Comment:
the outcome we're trying to assert is that user_op1 is executed, and userop2 is left in the mempool.

this can be achieved by:
`assert dump_mempool() == [user_op2]`

Can leave for now the test for user_op1, since we don't yet have a method `completed(userop)`, which should search for its log